linkcheck/logger/csvlog.py

```diff
@@ -119,12 +122,21 @@ def log_url (self, url_data):
         self.writerow(map(strformat.unicode_safe, row))
         self.flush()
 
+    def encode_row_s(self, row_s):
+        try:
+            return row_s.encode("utf-8", self.codec_errors)
+        except(TypeError, AttributeError):
+            return row_s
```

Comment:
This is one of those rare occasions where there should be a space in front of the `(`.

Why are you catching both TypeError and AttributeError?  On Python 3 I expect an AttributeError: str has no attribute encode.  On Python 2 I expect no error at all (the old code did not have any try/except).  When does the TypeError happen?

(Personally I would prefer a type check instead of try/except:

```python
     if isinstance(row_s, str):
        return row_s  # Python 3
    else:
        return row_s.encode(...)  # Python 2
```
but it's a matter of personal style, I think.)

Reply:
Don't know why both exceptions are being caught (didn't modify this one - maybe I merged two together).
I've implemented the isinstance alternative as suggested.